### PR TITLE
firmware: Initial RN2483 communication

### PR DESCRIPTION
**Pull request by rnestler**

### CODE REVIEW

File: platformio/src/main.cpp [hunk 1]
```diff
@@ -16,6 +16,32 @@ int send_command(I2C& i2c, uint8_t address, uint16_t command) {
...
+            ++t;
+            wait(0.1);
+            continue;
+        }
+        char c = rn2483.getc();
+        if (c == '\n') {
+            break;
+        }
+        answer.push_back(c);
+    }
```
Comment (dbrgn) [inline platformio/src/main.cpp:40]: Can we extract this into a separate helper function? You'll need this again later on, when figuring out whether connecting succeeded or not.

File: platformio/src/main.cpp [hunk 2]
```diff
@@ -16,6 +16,32 @@ int send_command(I2C& i2c, uint8_t address, uint16_t command) {
     return i2c.write(address, cmd, sizeof(cmd));
 }
 
+Serial rn2483(p13, p14);
+
+std::string sendcmd(const char* data) {
+    printf("> %s\n", data);
+    rn2483.printf("%s\r\n", data);
```
Comment (dbrgn) [inline platformio/src/main.cpp:24]: The `\r\n` _could_ be moved into a LINEBREAK constant or something like that :)

**Review by dbrgn — APPROVED** [review]

File: platformio/src/main.cpp (hunk 2 above)
Comment (rnestler) [reply platformio/src/main.cpp:24]: I'm not even sure if it needs `\r\n' or if '\n' would be enough.

File: platformio/src/main.cpp (hunk 1 above)
Comment (rnestler) [reply platformio/src/main.cpp:40]: Just the for loop to read the answer? would probably be smart yes. Also I'm not sure if allocating a `std::string` is a smart thing to do for every read.

File: platformio/src/main.cpp [hunk 3]
```diff
@@ -25,6 +51,24 @@ int main() {
...
+    /*
+    while(1) {
+        if(pc.readable()) {
+            rn2483.putc(pc.getc());
+        }
+        if(rn2483.readable()) {
+            pc.putc(rn2483.getc());
+        }
+    }
+    */
```
Comment (rnestler) [inline platformio/src/main.cpp:70]: I should probably remove this.

File: platformio/src/main.cpp (hunk 2 above)
Comment (dbrgn) [reply platformio/src/main.cpp:24]: As far as I remember it does need CRLF. And if it's a constant we can change it easily :)

File: platformio/src/main.cpp (hunk 1 above)
Comment (dbrgn) [reply platformio/src/main.cpp:40]: You could pass in a reference to a preallocated string, right?

Comment (rnestler) [reply platformio/src/main.cpp:40]: Yes, although there is no `mut&` in C++ so the code would look like it's passed as an input parameter :wink:

**Comment by rnestler** [comment]
Since this project moved to https://github.com/gfroerli/firmware there isn't really a point in keeping PRs open here is it?

**Comment by dbrgn** [comment]
Definitely. This is only here for archival purposes.

**Pull request closed without merging**